networking/eth2/src/test-support/java/tech/pegasys/artemis/networking/eth2/Eth2NetworkFactory.java

```diff
@@ -111,7 +134,16 @@ protected Eth2Network buildNetwork(final NetworkConfig config) {
         final P2PNetwork<?> network =
             new LibP2PNetwork(config, METRICS_SYSTEM, rpcMethods, peerHandlers);
 
-        return new Eth2Network(network, eth2PeerManager, eventBus, chainStorageClient);
+        NetworkConfig discoveryConfig =
+            new NetworkConfigBuilder()
+                .privateKey(config.getPrivateKey())
+                .peers(new ArrayList<>())
+                .networkInterface("127.0.0.1")
+                .listenPort(config.getListenPort() + 1)
+                .advertisedPort(config.getAdvertisedPort() + 1)
```

Comment:
We should be able to use the same ports because discovery should run over udp while the network runs over tcp.

Reply:
Good point, removed references to TCP in Eth2Discovery

Reply:
Sorry, just catching this comment.  We still want the `DiscoveryPeer` to be able to return the tcp port from its node record.  And we'll want to use the tcp port when [trying to connect](https://github.com/PegaSysEng/artemis/pull/1062/files#diff-d0fdc58ac989b3a46213a2d1aad190deR40) to the discovered peer with the `P2PNetwork`.  Its just that when we're spinning up nodes here we can use the same port number for both discovery (udp) and p2p networking (tcp).

Reply:
What about having the discovery interface and port as distinct fields from the libp2p interface and port? It allows running them separately as as microservices.

Reply:
I'd probably keep it simple for now until we decide we need discovery as a separate service. 🤷‍♀